firebase-common/src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -551,33 +533,13 @@ protected FirebaseApp(Context applicationContext, String name, FirebaseOptions o
             LibraryVersionComponent.create(FIREBASE_ANDROID, ""),
             LibraryVersionComponent.create(FIREBASE_COMMON, BuildConfig.VERSION_NAME),
             DefaultUserAgentPublisher.component());
-    publisher = componentRuntime.get(Publisher.class);
-  }
-
-  private static String getSharedPrefsName(String appName) {
-    return FIREBASE_APP_PREFS + appName;
-  }
-
-  private boolean readAutoDataCollectionEnabled() {
-    if (sharedPreferences.contains(DATA_COLLECTION_DEFAULT_ENABLED)) {
-      return sharedPreferences.getBoolean(DATA_COLLECTION_DEFAULT_ENABLED, true);
-    }
-    try {
-      PackageManager packageManager = applicationContext.getPackageManager();
-      if (packageManager != null) {
-        ApplicationInfo applicationInfo =
-            packageManager.getApplicationInfo(
-                applicationContext.getPackageName(), PackageManager.GET_META_DATA);
-        if (applicationInfo != null
-            && applicationInfo.metaData != null
-            && applicationInfo.metaData.containsKey(DATA_COLLECTION_DEFAULT_ENABLED)) {
-          return applicationInfo.metaData.getBoolean(DATA_COLLECTION_DEFAULT_ENABLED);
-        }
-      }
-    } catch (PackageManager.NameNotFoundException e) {
-      // This shouldn't happen since it's this app's package, but fall through to default if so.
-    }
-    return true;
+    dataCollectionConfigStorage =
+        new Lazy<>(
```

Comment:
I think I'm confused about the use of lazy in this context over just provider. 

Whats gained? Is it that lazy is memoized and we expect this provider to be called multiple times?

Reply:
Yeah, the key is that we want to memoize the value once it is created, but only want to create it when it is needed and not earlier